overlord/snapstate/aliases.go

```diff
@@ -177,29 +222,49 @@ func (m *SnapManager) doAlias(t *state.Task, _ *tomb.Tomb) error {
 			}
 			return fmt.Errorf("cannot %s alias %q for %q, no such alias", action, alias, snapName)
 		}
-		if aliasStatuses[alias] == newStatus {
-			// nothing to do
-			continue
-		}
-		beAlias := &backend.Alias{
-			Name:   alias,
-			Target: filepath.Base(aliasApp.WrapperPath()),
-		}
+		op := "nop"
 		switch newStatus {
 		case "enabled":
 			if aliasStatuses[alias] != "auto" {
+				op = "add"
```

Comment:
The logic here feels a lot less clear than it was originally. Can we merge these conditions and try to preserve the original structure with something like:
```
switch newStatus {
case "enabled", "auto":
    if aliasStatuses[alias] != "auto" && aliasStatuses[alias] != "enabled" {
            add = append(add, beAlias)
    }
...
}